packages/frontend-v2/pages/login.tsx

```diff
@@ -0,0 +1,135 @@
+import {
+  Button,
+  FormControl,
+  FormErrorMessage,
+  Input,
+  InputGroup,
+  Text,
+} from "@chakra-ui/react";
+import { API } from "@graduate/api-client";
+import { classValidatorResolver } from "@hookform/resolvers/class-validator";
+import { NextPage } from "next";
+import { useRouter } from "next/router";
+import { useState, useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { LoginStudentDto } from "../temp/dto-types";
+import { AxiosError } from "axios";
+
+const Login: NextPage = () => {
+  const [apiError, setApiError] = useState("");
+  const [renderSpinner, setRenderSpinner] = useState(true);
+  const router = useRouter();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+  } = useForm<LoginStudentDto>({
+    resolver: classValidatorResolver(LoginStudentDto),
```

Comment:
what does this do? not sure how much validation you really want to do here, because most of it should be handled via api errors anyway

Reply:
This will basically perform all the same validation that the backend has on the frontend. It ensures that the email is an email, and that the password provided is a string

Reply:
since we can reuse the dto classes with the class-validator decorators to do the same validation on the frontend that the backend does I think it's fine to validate on the frontend, saves us an api call for data that is obviously malformed without any additional logic needed

Reply:
you can use input type="email" to do email validation. and the password that is provided will always be a string, from the input. if it isn't, the UI should handle it

Reply:
fair fair, i'm okay with not using class-validator on the frontend and keeping validation simple as alex mentioned